src/main/java/net/mcreator/ui/workspace/selector/WorkspaceSelector.java

```diff
@@ -91,18 +93,23 @@ public WorkspaceSelector(@Nullable MCreatorApplication application, WorkspaceOpe
 				}
 			});
 
-		JPanel actions = new JPanel(new FlowLayout(FlowLayout.LEFT, 10, 10));
-		addWorkspaceButton(L10N.t("dialog.workspace_selector.new_workspace"), UIRES.get("addwrk"), e -> {
-			NewWorkspaceDialog newWorkspaceDialog = new NewWorkspaceDialog(this);
-			if (newWorkspaceDialog.getWorkspaceFile() != null)
-				workspaceOpenListener.workspaceOpened(newWorkspaceDialog.getWorkspaceFile());
-		}, actions);
+		JPanel actions = new JPanel(new BorderLayout(0, 6));
+
+		JButton newWorkspace = mainWorkspaceButton(L10N.t("dialog.workspace_selector.new_workspace"),
+				UIRES.get("addwrk"), e -> {
+					NewWorkspaceDialog newWorkspaceDialog = new NewWorkspaceDialog(this);
+					if (newWorkspaceDialog.getWorkspaceFile() != null)
+						workspaceOpenListener.workspaceOpened(newWorkspaceDialog.getWorkspaceFile());
+				});
+
+		actions.add("North", newWorkspace);
+		actions.add("Center", subactions);

```

Comment:
Maybe this should be wrapped in a scroll pane for cases when there are too many custom buttons? Or are these cases barely possible?

Reply:
The UI is self-centering, so new workspace button is moved upper if needed, if more than this many buttons are present, this dialog will be mess anyway and plugins really should not abuse this window with buttons